packages/slate-react/src/components/leaf.js

```diff
@@ -101,7 +109,16 @@ class Leaf extends React.Component {
     const leaf = this.renderText()
 
     return marks.reduce((children, mark) => {
-      const props = { editor, mark, marks, node, offset, text, children }
+      const props = {
+        editor,
+        mark,
+        marks,
+        node,
+        offset,
+        text,
+        children,
+        attributes,
+      }
```

Comment:
Can you just define `attributes` inline here instead of pulling it up above, since it's only used in one place?

Reply:
Fixed, thank you

Reply:
If we want to define attributes in `reduce` function; how about `data-slate-mark:index`; then we may be able to find leaf-mark by index in future development if necessary.